Lagt til fleire linters frå `lintr`-pakken

haugedal opened the pull request:
Køyrer no alle linterane som finst i `lintr`-pakken, utanom dei merka `deprecated` og ei handfull andre me ikkje ynskjer/treng.

haugedal committed:
Køyr alle linters (med nokre unntak), ikkje berre defaults

Argumentet tags = NULL gjer at alle linters
(utanom dei som er utdaterte) vert brukt.
Dei utdaterte vert ekskludert med standardverdien
exclude_tags = "deprecated".

haugedal committed:
Ikkje lint MASS:::profile.glm

MASS:::profile.glm er nødvendig i ein overgangsperiode
før funksjonen er flytta til stats-pakken i R versjon 4.4.0.

haugedal committed:
Kall eksporterte funksjonar med ::, ikkje :::

haugedal committed:
Bruk anyNA(x) i staden for any(is.na(x))

haugedal committed:
Fjerna unødvendig bruk av == i if()

haugedal committed:
Fjerna unødvendig bruk av if_else()

haugedal committed:
Bruk as.numeric(x) i staden for ifelse(x, 1, 0)

Dette fjernar òg bruken av ifelse() i staden for if_else().

haugedal committed:
Argumentet sep finst ikkje i paste0()

Det var berre tilfeldig at koden fungerte som tenkt.

haugedal committed:
Bruk str_flatten_comma(x) i staden for paste0(x, collapse = ", ")

haugedal committed:
Bruk str_flatten_comma(x) i staden for str_c(x, collapse = ", ")

haugedal committed:
Lag adresser med file.path(...) i staden for paste0(x, "/", y, "/", z)

haugedal committed:
Ikkje bruk hermeteikn i namngjevne argument til funksjonar

haugedal committed:
Bruk stringr::fixed() i staden for statiske regulære uttrykk

Det er raskare.

haugedal committed:
Bruk if (A) x else y i staden for det mindre lettleste if (!A) y else x

I alle fall i enkle if/else-uttrykk.

haugedal committed:
Bruk anyDuplicated(x, ...) > 0 i staden for any(duplicated(x), ...)

anyDuplicated() exists as a replacement for any(duplicated(.)),
which is more efficient for simple objects,
and is at worst equally efficient.
Therefore, it should be used in all situations instead of the latter.

haugedal committed:
Skriv alltid ut 0 før komma (0.1 vs. .1), det er meir lettlest

While .1 and 0.1 mean the same thing,
the latter is easier to read due to the small size of the '.' glyph.

haugedal committed:
Bruk sort(x, na.last = TRUE) i staden for x[order(x)]

Det er raskare (og meir lettlest?).

haugedal committed:
Bruk lengths() til å finna lengda til alle element i ei liste

lengths() is a function that was added to base R in version 3.2.0
to get the length of each element of a list.
It is equivalent to sapply(x, length), but faster and more readable.

haugedal committed:
Unngå bruk av setwd() og options()

haugedal committed:
Bruk map_lgl() i staden for sapply()

haugedal committed:
Ikkje lint bruk av library() i vignettar og testthat.R

haugedal committed:
Ikkje lint bruk av options() i tema-kvalreg.R

haugedal committed:
Korrekt syntaks for nolint-kommentarar

haugedal committed:
Argument utan standardverdiar skal koma før argument med standardverdiar

haugedal committed:
Køyr etterfølgande kall av stopifnot() i eitt

stopifnot() accepts any number of tests,
so sequences like ⁠stopifnot(x); stopifnot(y)⁠ are redundant.
Ditto for tests using assertthat::assert_that() without specifying ⁠msg=⁠.

haugedal committed:
Bruk stopifnot(A, B) i staden for stopifnot(A &&B)

Det er meir lettlest, gjev betre feilmeldingar.
Det same gjeld dplyrr::filter().

haugedal committed:
Unødvendig bruk av anonym funksjon

haugedal committed:
Plassert på éi linje

haugedal committed:
Bruk paste0() i staden for paste()

haugedal committed:
Godta bruk av anonyme funksjonar i purrr-funksjonar

Frå purrr:
We now generally recommend against using ... to pass
additional (constant) arguments to .f.
Instead use a shorthand anonymous function:

# Instead of
x |> map(f, 1, 2, collapse = ",")
# do:
x |> map(\(x) f(x, 1, 2, collapse = ","))

Sjå issue på GitHub om dette i unnecessary_lambda_linter():
https://github.com/r-lib/lintr/issues/1871

haugedal committed:
Bruk !any(x) i staden for all(!x) og omvendt

any(!x) is logically equivalent to !any(x);
ditto for the equivalence of all(!x) and !any(x).
Negating after aggregation only requires inverting
one logical value, and is typically more readable.

haugedal committed:
Bruk == til å matcha skalarar av lengd 1, ikkje %in%

Set òg na.rm = TRUE sidan == tek vare på NA-verdiar,
medan %in% ikkje gjer det, og me ikkje vil ha NA inne i any().

haugedal committed:
Ikkje lint med absolute_path_linter() for denne kodesnutten

Her er hardkoda filadresse berre eit eksempel,
og ikkje kode som vert køyrt. Har òg sett echo=2:11 slik at
nolint-kommentarane ikkje vert med i den kompilerte vignetten.

haugedal committed:
Retta feil i logiske testar

Testane skal vera if(length(x) > 0), ikkje if(length(x > 0)).
«Tilfeldigvis» funka koden som tenkt slik han stod likevel,
sidan if(0) ikkje vil køyra, og if(1), if(2) osv. vil køyra.

haugedal committed:
Ikkje samanlikn klassar med ==

Usage like class(x) == "character" is prone to error
since class in R is in general a vector.
The correct version for S3 classes is inherits():
inherits(x, "character").
Often, class k will have an is. equivalent,
for example is.character() or is.data.frame().

haugedal committed:
Bruk heller if(!(A && B)) enn if(!A || !B)

haugedal committed:
Ignorer testthat.R i .lintr-fila i staden for med kommentarar

haugedal committed:
Bruk expect_identical() i staden for expect_equal() om mogleg

expect_equal() should only be used when expect_identical() is
inappropriate, i.e., when x and y need only be numerically equivalent
instead of fully identical (in which case, provide the
tolerance=⁠ argument to expect_equal() explicitly).
This also applies when it's inconvenient to check full equality
(e.g., names can be ignored, in which case ignore_attr = "names"
should be supplied to expect_equal().

haugedal committed:
Unngå bruk av c() men ingen eller berre eitt element

haugedal committed:
Fjerna unødvendig bruk av as.logical()

haugedal committed:
Bruk expect_named(x, namn) i staden for expect_identical(names(x), namn)

testthat::expect_named() exists specifically for testing the names()
of an object. testthat::expect_equal() can also be used for such tests,
but it is better to use the tailored function instead.

haugedal committed:
Bruk expect_true() og expect_false()

testthat::expect_true() and testthat::expect_false() exist
specifically for testing the TRUE/FALSE value of an object.
testthat::expect_equal() and testthat::expect_identical()
can also be used for such tests,
but it is better to use the tailored function instead.

haugedal committed:
Bruk expect_gt(x, y) i staden for expect_true(x > y)

haugedal committed:
Fjerna overflødig komma

haugedal committed:
Bruk str_starts(x, "...") i staden for str_detect(x, "^...")

haugedal committed:
Flytta kommentar for å unngå for lang linje

haugedal committed:
Lint-unntak

haugedal committed:
Oppdatert til Roxygen versjon 7.3.1

haugedal committed:
Bruk undesirable_operator_linter() i staden for heimesnekra assignment-linter

Kan då òg fjerna lint-unntaka for <<- og :=.

haugedal committed:
Ikkje bruk todo_comment_linter()

haugedal committed:
Retta kodestil

haugedal committed:
Oppdatert Roxygen til versjon 7.3.2

haugedal committed:
Fått testar til å funka på macOS

Oppdaga ved automatisk køyring av testar på GitHub med ulike OS
at desse feila på macOS (men ikkje på Windows og Linux).
Dei feila med «Objects equal but not identical»,
som i desse tilfella kan fiksast med å gjera om
expect_identical(...) til expect_equal(..., tolerance = x).

orjanls commented on R/figur-funksjoner.R:
Bør også bytte rekkefølgen på `nevner` og `figtype` i `@description`.

haugedal replied:
Har no gjort tilsvarande endring i andre funksjonar der rekkefylgja på argument vart endra òg.

haugedal commented:
Ser ut som codecov ikkje køyrer som det skal. Testen er grøn, men det står:
```
"Rate limit reached. Please upload with the Codecov repository upload token to resolve issue.
```

Etter litt kjapp googling kan det sjå ut som at «tokenless uploading» ikkje er støtta lenger, og at det var det me brukte før? Skal sjå om eg kan få det til å funka igjen.

haugedal committed:
Bytt rekkefylgje på argumenta i hjelpeteksten òg

Argumenta i funksjonen vart tidlegare flytta slik at
alle argument utan standardverdi kjem fyrst.
